lib/constellation.rb

```diff
@@ -1,58 +1,31 @@
-require 'net/http'
+require "graphql/client"
+require "graphql/client/http"
 module Constellation
-  class HttpException < StandardError
+  class GraphQLException < StandardError
   end
-  HTTP_ERRORS = [
-    Errno::ETIMEDOUT,
-    Timeout::Error,
-    Net::OpenTimeout,
-    Net::ReadTimeout,
-    EOFError,
-    Errno::ECONNABORTED,
-    Errno::ECONNREFUSED,
-    Errno::ECONNRESET,
-    Errno::EHOSTDOWN,
-    Errno::EHOSTUNREACH,
-    Errno::EINVAL,
-    Errno::ENETUNREACH,
-    SocketError,
-    OpenSSL::SSL::SSLError,
-    Net::HTTPBadResponse,
-    Net::HTTPHeaderSyntaxError,
-    Net::ProtocolError,
-    Zlib::GzipFile::Error,
-  ]
 
-  def self.create_rsvp!(rsvp_params)
-    return unless constellation_enabled?
-    uri = URI.parse("#{app}/rsvps")
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = true
-    req = Net::HTTP::Post.new(uri.path, headers)
-    req.body = { rsvp: rsvp_params.merge(total_count: true) }.to_json
-    response = http.request(req)
-    raise Constellation::HttpException, response.body unless response.code == '201'
-    return JSON.parse(response.body), response['X-Total-Count'].to_i
-  rescue *HTTP_ERRORS => e
-    raise Constellation::HttpException, e.message
-  end
-
-  def self.constellation_enabled?
-    app && token
+  HTTP = GraphQL::Client::HTTP.new("#{RsvpRails.config[:constellation_url]}/api/graphql/") do
+    def headers(_)
+      { 'Authorization' => "Bearer #{RsvpRails.config[:constellation_jwt_token]}" }
+    end
   end
+  Schema = GraphQL::Client.load_schema(HTTP)
+  Client = GraphQL::Client.new(schema: Schema, execute: HTTP)
 
-  def self.app
-    RsvpRails.config[:constellation_url]
-  end
+  CreateRsvpMutation = Client.parse <<-GRAPHQL
+    mutation($input: createRsvpInput!) {
```

Comment:
Took a little while to get this right, `Client.parse`, and also the `createRsvpInput!` type.

Reply:
What was it? The client API? The syntax of the mutation? In case of the latter, I would really like (you) to try this Ruby code setup with [graphql-for-vscode](https://github.com/kumarharsh/graphql-for-vscode#whats-in-the-box). Also, because moving to a Ruby DSL version means we’d probably lose all benefits of a static query language like GraphQL’s.